lib/registrant.rb

```diff
@@ -0,0 +1,22 @@
+class Registrant
+  attr_reader :name, 
+              :age, 
+              :permit
+  attr_accessor :license_data
+  def initialize(name, age, permit = false)
+    @name = name
+    @age = age
+    @permit = permit
+    @license_data = { written: false,
+        license: false,
+        renewed: false }
+  end
+
+  def permit?
+    @permit
+  end
+
+  def earn_permit
+    @permit = true
+  end
+end
```

Comment:
This class is great. The only thing I will add is that you've defined an `attr_accessor` for `@license_data`, which allows direct modification of the license_data hash from outside the class. This is only necessary if you want to allow `@license_data` to point to something other than the hash it is currently pointing to.

Because `@license_data` is pointing to a hash and hashes can be updated without the `attr_writer`, line 5 can be removed and you would need to add `:license_data` to the `attr_reader` list.